plugins/activity-resources/src/components/utils.ts

```diff
@@ -164,11 +166,26 @@ function getHiddenAttrs (client: TxOperations, _class: Ref<Class<Doc>>): Set<str
 }
 
 export async function getValue (client: TxOperations, m: AttributeModel, utx: any): Promise<any> {
-  const val = utx[m.key]
-
-  if (client.getHierarchy().isDerived(m._class, core.class.Doc) && typeof val === 'string') {
-    // We have an reference, we need to find a real object to pass for presenter
-    return await client.findOne(m._class, { _id: val as Ref<Doc> })
+  async function getAllRealValues (values: any[]): Promise<any[]> {
+    return client.getHierarchy().isDerived(m._class, core.class.Doc) &&
+      values.every((value) => typeof value === 'string')
+      ? await client.findAll(m._class, { _id: { $in: values } })
+      : values
+  }
+  const value = {
+    set: utx[m.key],
+    added: utx.$push?.[m.key],
+    removed: utx.$pull?.[m.key]
+  }
+  if (value.set !== undefined) {
+    ;[value.set] = await getAllRealValues([value.set])
   }
-  return val
+  if (value.added !== undefined) {
+    value.added = await getAllRealValues(Array.isArray(value.added.$each) ? value.added.$each : [value.added])
+  }
+  if (value.removed !== undefined) {
```

Comment:
Removed will not be located from snapshot storage and need to be restored from transactions. Please see how document it restored from transactions in activity.

Reply:
It is only for `$pull` operation in `TxUpdateDoc`
But if `attachedDoc` was deleted, presenter will not be shown for all `txes`